ansible/roles/prometheus-server/tasks/main.yml

```diff
@@ -0,0 +1,85 @@
+---
+
+- name: ensure {{ prometheus_group }} group exists
+  become: yes
+  group:
+    name: "{{ prometheus_group }}"
+    system: yes
+
+- name: ensure {{ prometheus_user }} user exists
+  become: yes
+  user:
+    name: "{{ prometheus_user }}"
+    group: "{{ prometheus_user }}"
+    system: yes
+
+- name: check existing volume mounts
+  set_fact:
+    volume_mounted: "{{ ansible_mounts | selectattr('mount', 'match', prometheus_data_dir) | list | length > 0 }}"
+
+- name: configure volume mounts
+  include_role:
+    name: mounted-volume
+  when:
+    - not volume_mounted
+    - volume_mounts is defined
+    - volume_mounts != ""
+
+- name: prometheus directory structure
+  file:
+    state: "directory"
+    path: "{{ item }}"
+    owner: "{{ prometheus_user }}"
+    group: "{{ prometheus_group }}"
+  with_items:
+    - "{{ prometheus_root_dir }}"
+    - "{{ prometheus_dist_dir }}"
+    - "{{ prometheus_config_dir }}"
+    - "{{ prometheus_data_dir }}"
+
+- name: check if prometheus tarball exists locally
+  stat:
+    path: "{{ prometheus_dist_dir }}/{{ prometheus_tar_gz }}"
+  register: prometheus_tar
+
+- name: download prometheus tarball when not exists
+  get_url:
+    url: "{{ prometheus_tar_gz_url }}"
+    dest: "{{ prometheus_dist_dir }}/{{ prometheus_tar_gz }}"
+    validate_certs: "no"
+  when: not prometheus_tar.stat.exists
+
+- name: extract tarball
+  unarchive:
+    src: "{{ prometheus_dist_dir }}/{{ prometheus_tar_gz }}"
+    dest: "{{ prometheus_dist_dir }}"
+    copy: "no"
+    creates: "{{ prometheus_dist_dir }}/prometheus-{{ prometheus_version }}.linux-amd64/prometheus"
+
+- name: cleanup tarball
+  file:
+    state: absent
+    path: "{{ prometheus_dist_dir }}/{{ prometheus_tar_gz }}"
+
+- name: symlink to prometheus server folder
+  file:
+    src: "{{ prometheus_dist_dir }}/prometheus-{{ prometheus_version }}.linux-amd64"
+    dest: "{{ prometheus_server_dir }}"
+    state: link
+
+- name: default prometheus config file
+  copy:
+    src: "prometheus.yml"
+    dest: "{{ prometheus_config_dir }}/prometheus.yml"
+    mode: "0644"
+
+- name: configure prometheus service
+  template:
+    src: "etc/systemd/system/prometheus.service.j2"
+    dest: "/etc/systemd/system/prometheus.service"
+
+- name: set prometheus service state
+  service:
+    name: "prometheus"
+    state: "started"
+    enabled: "yes"
```

Comment:
This task is redundant as we want control of services to be performed on a per-environment basis (i.e. if we want to disable service X for a given environment we don't want to be forced to provision an entirely new AMI to do so), and also because starting the service on the Packer build instance may have unintended side effects. Both operations can be performed using cloud-init on first instance startup as [shown here](https://github.com/companieshouse/smartpan-terraform/blob/2b255f6abb07a1c9b1a628c7f689ebff450b39db/groups/smartpan/module-smartpan/cloud-init/files/bootstrap-commands.yml#L2).

Reply:
Yeah, I see what you mean. I kind-of liked the setup where you could spin up an instance from the AMI and without having to do any other config, you had a working Prometheus environment. I did check for anything weird due to the service starting during the build and didn't spot anything. But I do take your point re possible unknown outcomes or unexpected data in the AMI. Removed now.
Will pass the details on to Craig to add to the Terraform cloud-init config.